docs/source/cuda._sanitizer.rst

```diff
@@ -0,0 +1,95 @@
+.. currentmodule:: torch.cuda._sanitizer
+
+torch.cuda._sanitizer
+=====================
+
+Overview
+--------
+
+.. automodule:: torch.cuda._sanitizer
+
+
+Usage
+------
+
+Here is an example of a simple synchronization error in PyTorch:
+
+::
+
+    import torch
+
+    a = torch.rand(4, 2, device="cuda")
+
+    with torch.cuda.stream(torch.cuda.Stream()):
+        torch.mul(a, 5, out=a)
+
+The ``a`` tensor is initialized on the default stream and, without any synchronization
+methods, modified on a new stream. When this script is run on the commandline with:
+::
+
+    TORCH_CUDA_SANITIZER=1 python example_error.py
+
+the following output is printed by CSAN:
+
+::
+
+    ============================
+    CSAN detected a possible data race on tensor with data pointer 139719969079296
+    Access by stream 94646435460352 during kernel:
+    aten::mul.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+    writing to argument: self, out, output
+    With stack trace:
+        File "pytorch/torch/cuda/_sanitizer.py", line 364, in _handle_kernel_launch
+            stack_trace = traceback.StackSummary.extract(
+        ...
+        File "example_error.py", line 6, in <module>
+            torch.mul(a, 5, out=a)
```

Comment:
I thought in Python the innermost frame is printed last, whereas here it comes first. Also by default traceback indents lines by two spaces, not four. Is this literally a verbatim output of CSAN, or did you build it by hand?

Reply:
This is a _slightly modified_ version of the verbatim output of CSAN, with the frames in the middle abbreviated for clarity. The indentation is a mistake on my part, I will fix it.

The order of the frames is just as CSAN prints it, which is how [`StackSummary`](https://docs.python.org/3/library/traceback.html#traceback.StackSummary) formats it.

Reply:
I've reversed the stored `StackSummary` objects.